src/validate.ts

```diff
@@ -0,0 +1,18 @@
+import { ZodError, ZodType } from 'zod'
+import { ResponseError } from './types'
+
+export function validate<T extends ZodType>(obj: any, schema: T): void {
+  try {
+    schema.parse(obj)
+  } catch (err) {
+    if (err instanceof ZodError) {
+      throw new ResponseError(
+        `Error validating object with schema ${
+          schema.description
+        }. ${JSON.stringify(err.format()._errors)}`,
```

Comment:
This would only include top level errors. We may want to include the whole thing if we wanted field specific errors. https://github.com/colinhacks/zod/blob/master/ERROR_HANDLING.md#formatting-errors